test/hotspot/jtreg/serviceability/jvmti/vthread/ToggleNotifyJvmtiTest/libToggleNotifyJvmtiTest.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <cstdlib>
+#include <cstring>
+#include <jvmti.h>
+#include "jvmti_common.h"
+
+extern "C" {
+
+static jvmtiEnv *jvmti;
+static volatile int vthread_started_cnt = 0;
+static volatile int vthread_ended_cnt = 0;
+static volatile int thread_started_cnt = 0;
+static volatile int thread_ended_cnt = 0;
+
+static jrawMonitorID agent_lock = NULL;
+static volatile jboolean agent_started = JNI_FALSE;
+
+void JNICALL VirtualThreadStart(jvmtiEnv *jvmti, JNIEnv* jni, jthread thread) {
+  if (!jni->IsVirtualThread(thread)) {
+    fatal(jni, "Failed: expected to be virtual thread");
+  }
+  RawMonitorLocker agent_locker(jvmti, jni, agent_lock);
+
+  vthread_started_cnt++;
```

Comment:
Wouldn't it be better to use std::atomic<int> instead RawMonitorLocker here to reduce sync time?

Reply:
This RawMonitorLocker is not on a critical path.
It can be useful to sync print statements when tracing is needed.
I feel that in order to get a full advantage of it we need to do this for many tests in our test base.
I can convert it to std::atomic if you think it is important.